### PR TITLE
Api implementation

### PR DESCRIPTION
**Pull request by r-Dev03**
Fixes for issue #4 ready for merge.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-const server = http.createServer((req, res) =>{
+const server = http.createServer((req, res) => {
     // if protocol is undefined, default to http for now
     const base = ((req.protocol)? req.protocol : "http") + '://' + req.headers.host + '/'
     const fullURL =  new url.URL(req.url, base)
     const page = fullURL.pathname
     const params = fullURL.searchParams
     if (page === "/api"){
         // TODO: handle api here
+        jsonSend(undefined, params, res)
```
Comment (MichaelPachec0) [inline server.js:23]:
```suggestion
        // TODO
```
This function is not in the code anymore, should be deleted and replaced with a comment for now.

File: server.js [hunk 2]
```diff
@@ -59,22 +63,33 @@ function fileExists(path) {
...
+function jsonSend(err, params, res) {
+    if (err) {
+        console.log('Something went wrong...');
+        console.dir(err);
+        return;
+    }
+    let cType = fileTypes["json"]
+    res.writeHead(200, { 'Content-Type': cType });
+    _parse(params).then((data) => res.end(data))
+}
```
Comment (MichaelPachec0) [inline server.js:94]:
```suggestion
```
as the comment mentions, this should not be committed yet

File: api.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      ret.push(rand)
+    }
+    return ret
+  })
+  const ret = {
+    index: shuffling_index,
+    data: json
+  }
+  return ret
+}
```
Comment (MichaelPachec0) [inline api.js:34]:
```suggestion
}

```
need an extra newline here.

File: .gitignore [hunk 4]
```diff
@@ -1 +1,2 @@
-node_modules
\ No newline at end of file
+node_modules
+.env
```
Comment (MichaelPachec0) [inline .gitignore:2]:
```suggestion
.env

```
newline needed here

File: package.json [hunk 5]
```diff
@@ -12,6 +12,8 @@
   "type": "module",
   "license": "MIT",
   "dependencies": {
-    "figlet": "^1.5.2"
+    "figlet": "^1.5.2",
+    "node-fetch": "^3.2.4",
+    "dotenv": "^16.0.1"
   }
 }
```
Comment (MichaelPachec0) [inline package.json:19]:
```suggestion
}

```
here as well

**Review by MichaelPachec0 — CHANGES_REQUESTED** [review]

**Review by MichaelPachec0 — APPROVED** [review]
LGTM

**Review by jericashall — APPROVED** [review]

**Comment by MichaelPachec0** [comment]
for now the glue to send json is not committed yet, but that will be next thing to commit. It works but i need to clean it up and integrate it with the rest of the code.

**Pull request merged**